dotenv-linter/src/cli/options.rs

```diff
@@ -1,48 +1,62 @@
-use crate::common::LintKind;
+use crate::{
+    common::LintKind,
+    schema::{self, DotEnvSchema},
+};
 use clap::ArgMatches;
 use std::path::PathBuf;
-
-pub struct CheckOptions<'a> {
+#[derive(Default)]
+pub struct CliOptions<'a> {
```

Comment:
Since there is no need to pass the whole `CliOptions` to  the `checks::run()` function, I suggest returning everything as it was 🙂 Except for adding a new `schema` field to the `CheckOptions` structure.

Reply:
The schema is needed by fix too - thats what made me do the unifications, fix calls check first.

Reply:
The scheme is not needed for the `fix` command, because in any case, we cannot fix the warnings about the violation of the scheme. So, just pass `None` as schema to the `fix` function.

Reply:
@mgrachev  schema can be used by both fix and compare.

- fix can remove (or better, comment out) lines that should not be there
- compare can use the schema , 123.0, 123 and 0123 are all the same if type is double